CrowdStrike/crowdstrike_telemetry/crowdstrike.py

```diff
@@ -0,0 +1,141 @@
+"""Contains connector, configuration and module."""
+import asyncio
+from functools import cached_property
+from gzip import decompress
+from typing import Any, Dict, List
+
+import orjson
+from loguru import logger
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+from sekoia_automation.module import Module
+
+from aws.s3 import S3Configuration, S3Wrapper
+from aws.sqs import SqsConfiguration, SqsWrapper
+
+from .schemas import CrowdStrikeEventSchema
+
+
+class CrowdStrikeTelemetryConfig(DefaultConnectorConfiguration):
+    """Configuration for CrowdStrikeTelemetryConnector."""
+
+    aws_secret_access_key: str
+    aws_access_key_id: str
+    queue_name: str
+    aws_region: str
+
+    chunk_size: int | None = None
+    frequency: int | None = None
+    delete_consumed_messages: bool | None = None
+    is_fifo: bool | None = None
+
+
+class CrowdStrikeTelemetryModule(Module):
+    """CrowdStrikeTelemetryModule."""
+
+    configuration: CrowdStrikeTelemetryConfig
+
+
+class CrowdStrikeTelemetryConnector(Connector):
+    """CrowdStrikeTelemetryConnector class to work with logs."""
+
+    name = "CrowdStrikeTelemetryConnector"
+    module: CrowdStrikeTelemetryModule
+
+    @cached_property
+    def sqs_wrapper(self) -> SqsWrapper:
+        """
+        Get SQS wrapper.
+
+        Returns:
+            SqsWrapper:
+        """
+        config = SqsConfiguration(**self.module.configuration.dict(exclude_unset=True, exclude_none=True))
+
+        return SqsWrapper(config)
+
+    @cached_property
+    def s3_wrapper(self) -> S3Wrapper:
+        """
+        Get SQS wrapper.
+
+        Returns:
+            SqsWrapper:
+        """
+        config = S3Configuration(**self.module.configuration.dict(exclude_unset=True, exclude_none=True))
+
+        return S3Wrapper(config)
+
+    async def get_crowdstrike_events(self) -> List[Dict[Any, Any]]:
+        """
+        Run CrowdStrikeTelemetry.
+
+        Returns:
+            list[dict]:
+        """
+        with self.sqs_wrapper.receive_messages(delete_consumed_messages=True) as messages:
+            validated_sqs_messages = [CrowdStrikeEventSchema.parse_raw(content) for content in messages]
+
+            s3_data_list = await asyncio.gather(
+                *[
+                    self.process_s3_file(file.path, record.bucket)
+                    for record in validated_sqs_messages
+                    for file in record.files
+                ]
+            )
+
+            # We might have list of lists at this point we should flatten it
+            result = []
+            for record in s3_data_list:
+                if isinstance(record, list):
+                    for item in record:
+                        result.append(item)
+                else:
+                    result.append(record)
+
+            logger.info("Found {0} records to process".format(len(result)))
+
+            return result
```

Comment:
Where does the result go? I find a call of `get_crowdstrike_events` in the async loop, but the result is not used.